### PR TITLE
refactor: convert `type` properties to use enums

### PR DESCRIPTION
**Pull request by iamacook**
As mentioned in https://github.com/safe-global/safe-client-gateway/pull/910#discussion_r1415506401, this implements enums and updates tests/builders accordingly for:

- `ExecutionDetails['type']`
- `ExecutionInfo['type']`
- `QueuedItem['type']`
- `SettingsChange['type']`
- `Transfer['type']`
- `TransactionInfo['type']`

### CODE REVIEW

File: src/routes/transactions/entities/__tests__/transfer-transaction-info.builder.ts [hunk 1]
```diff
@@ -18,6 +20,6 @@ export function transferTransactionInfoBuilder(): IBuilder<TransferTransactionIn
     )
     .with('transferInfo', {
       ...erc20TransferBuilder().build(),
-      type: 'ERC20_TRANSFER',
```
Comment (iamacook) [inline src/routes/transactions/entities/__tests__/transfer-transaction-info.builder.ts:21]: This was incorrect. `ERC20_TRANSFER` is returned by the Transaction Service, but the `transferInfo` should return `ERC20`.

File: src/routes/transactions/mappers/transfers/transfer-info.mapper.ts [hunk 2]
```diff
@@ -19,10 +19,6 @@ import { Transfer } from '@/routes/transactions/entities/transfers/transfer.enti
 
 @Injectable()
 export class TransferInfoMapper {
-  private static readonly ERC20_TRANSFER = 'ERC20_TRANSFER';
-  private static readonly ERC721_TRANSFER = 'ERC721_TRANSFER';
-  private static readonly ETHER_TRANSFER = 'ETHER_TRANSFER';
```
Comment (iamacook) [inline src/routes/transactions/mappers/transfers/transfer-info.mapper.ts:24]: These are not used.

**Review by fmrsabino — APPROVED** [review]
Great job! 👏

File: src/routes/transactions/entities/__tests__/transfer-transaction-info.builder.ts (hunk 1 above)
Comment (iamacook) [reply src/routes/transactions/entities/__tests__/transfer-transaction-info.builder.ts:21]: I agree. Luckily the enum conversaion acted as a defacto test and caused a type error.

**Pull request merged**